### ISSUE TITLE
2.2: TextFormField's Suffix , Prefix, Icon and Label don't respond to  Primary Color's Change.

### ISSUE THREAD

**Issue by berkbb** [flutter/flutter#83002]
Hi,

Changing the primary color for main theme  in main. dart linke this , affects TextFormField's Suffix , Prefix, Icon and Label focus color normally in previous versions:

```
theme: ThemeData(
        visualDensity: VisualDensity.adaptivePlatformDensity,

        primaryColor:
            Color.fromRGBO(181, 82, 158, 1), // https://www.colorhexa.com/b5529e
),
```
My example TextFormField is:

```
TextFormField(

                                    controller: idController,
                                    // The validator receives the text that the user has entered.
                                    validator: (value) {
                                      if (value == null ||
                                          value.isEmpty ||
                                          int.tryParse(value) == null ||
                                          value.length != 11) {
                                        return 'Hello.';
                                      }
                                      return null;
                                    },
                                    obscureText: obscureID,
                                    inputFormatters: [
                                      FilteringTextInputFormatter.digitsOnly
                                    ],
                                    maxLength: 11,
                                    keyboardType: TextInputType.number,
                                    textInputAction: TextInputAction.next,
                                    onEditingComplete: () =>
                                        node.nextFocus(), // Move focus to next
                                    decoration: InputDecoration(
                                      
                                      icon: Icon(Icons.badge),
                                   
                                      labelText: 'Hello',
                                      suffixIcon: Tooltip(

                                        message: 'Show Hello',
                                        child: IconButton(
                                          onPressed: () {
                                            setState(() {
                                              obscureID = !obscureID;
                                            });
                                          },
                                          icon: Icon(Icons.remove_red_eye),
                                        ),
                                      ),
                                    ),
                                  ),
```

However, always blue color comes while focusing, I can change border color of field with modifying the InputDecorationTheme like this, but it not run on Suffix , Prefix, Icon and Label focused colors:

```
   inputDecorationTheme: InputDecorationTheme(  focusedBorder:OutlineInputBorder(
          borderSide: const BorderSide(color:    Color.fromRGBO(181, 82, 158, 1),

               width: 2.0),

        ),hoverColor: Color.fromRGBO(226,226, 226, 0.3),
          filled: true,
          fillColor: Colors.white,
          border: OutlineInputBorder(),



        ),
```
Screenshots (You can see the blue color, I expected that a purple-ish color based on primary color)

![image](https://user-images.githubusercontent.com/23173536/118959408-f8fe0100-b96a-11eb-9f15-90403c2c0298.png)
![image](https://user-images.githubusercontent.com/23173536/118959422-fbf8f180-b96a-11eb-949c-f26150b83709.png)

Flutter Analyze (Caused from web file picker):
```
 info • Don't import implementation files from another package • lib/generated_plugin_registrant.dart:7:8 • implementation_imports

1 issue found. (ran in 3.7s)
```
Flutter Doctor:
```

[✓] Flutter (Channel stable, 2.2.0, on Mac OS X 10.15.7 19H1030 darwin-x64, locale tr-TR)
    • Flutter version 2.2.0 at /Users/berkbabadogan/FlutterSDK
    • Framework revision b22742018b (5 days ago), 2021-05-14 19:12:57 -0700
    • Engine revision a9d88a4d18
    • Dart version 2.13.0

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/berkbabadogan/AndroidSDK
    • Platform android-S, build-tools 30.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.56.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.22.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • Mac OS X 10.15.7 19H1030 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 90.0.4430.212

• No issues found!

```
Bug.zip at [here](https://github.com/flutter/flutter/files/6514831/bug.zip).

**Comment by famazi111** [comment]
You have to use colorscheme property now.

https://flutter.dev/docs/release/breaking-changes/theme-data-accent-properties


```
colorScheme: ColorScheme.fromSwatch(
      primarySwatch: MaterialColor(0xffbd1b33, corPrimariaSwatch),
      brightness: Brightness.light,
    ),
```

**Comment by darshankawar** [comment]
@berkbb 
Check above link with breaking changes and see if it answers your question as to why you are seeing such behavior.

Thanks.

**Comment by berkbb** [comment]
Hi,

I saw the breaking changes @famazi111 , but "Landed in version: 2.3.0-0.1.pre , In stable release: not yet" written bottom of document. I'm using 2.2 stable, so I thought it is not applicable for me :)


Now, I created a material Color item bottom:
```

///Primary swatch list. Needed for focus colors.
  final MaterialColor primaryColorSwatch = const MaterialColor(
    0xffb5529e, // 0% comes in here, this will be color picked if no shade is selected when defining a Color property which doesn’t require a swatch.
    const <int, Color>{
      50: const Color(0xff813770 ),//10%
      100: const Color(0xff743164),//20%
      200: const Color(0xff662c58),//30%
      300: const Color(0xff58264c),//40%
      400: const Color(0xff4a2040),//50%
      500: const Color(0xff3d1a35),//60%
      600: const Color(0xff2f1429),//70%
      700: const Color(0xff210e1d),//80%
      800: const Color(0xff130811),//90%
      900: const Color(0xff060205),//100%
    },
  );
```

Added it to primarySwatch:

`primarySwatch: primaryColorSwatch,`

These snippets solved my issue. However, I added your solution also for test:

```
  primaryColor: Color.fromRGBO(181, 82, 158, 1),
        // https://www.colorhexa.com/b5529e,

        accentColor: Color.fromRGBO(82, 181, 105, 1),
        // https://www.colorhexa.com/52b569


        primarySwatch: primaryColorSwatch,

         colorScheme: ColorScheme.fromSwatch(
           primarySwatch:primaryColorSwatch,
           brightness: Brightness.dark,
         ),

```

After adding your code suggestion , it fails.

```
Assertion failed: file:///Users/berkbabadogan/FlutterSDK/packages/flutter/lib/src/material/theme_data.dart:316:12
colorScheme?.brightness == null || brightness == null || colorScheme!.brightness == brightness
is not true

```
![Ekran Resmi 2021-05-20 18 47 10](https://user-images.githubusercontent.com/23173536/119009472-d0dac600-b99b-11eb-9537-c295ac05f8b6.png)

I removed suggested code snippet, it runs perfectly.

![Ekran Resmi 2021-05-20 18 48 05](https://user-images.githubusercontent.com/23173536/119009572-ee0f9480-b99b-11eb-913e-c17f31787524.png)

Apart from  [https://flutter.dev/docs/release/breaking-changes/theme-data-accent-properties](https://flutter.dev/docs/release/breaking-changes/theme-data-accent-properties), I think this is 2.2 stable issue @darshankawar.

**Comment by darshankawar** [comment]
@berkbb 
Thanks for digging in. Can you provide a complete code sample in a single file, that we can verify to see if this is a regression ?

**Comment by berkbb** [comment]
I will create a new project with my theme code @darshankawar . You can observe the error by commenting the primary swatch.

**Comment by berkbb** [comment]
[Example Project is here.](https://github.com/flutter/flutter/files/6520795/example_bug.zip)

**Comment by famazi111** [comment]
> [Example Project is here.](https://github.com/flutter/flutter/files/6520795/example_bug.zip)
> 

Commenting primary swatch and using my example the error occurs.

Changing the brightness in colorScheme to light instead of dark it works.

**Comment by darshankawar** [comment]
@berkbb 
Thanks for the sample project. I ran and verified it on stable 2.0.6 and 2.2.0 on mobile and web and seems to be happening consistently on both, which indicates it's not specific to web, but looks like a framework level issue.

1. Stable (2.0.6) : mobile:

<img width="340" alt="Screenshot 2021-05-21 at 6 04 21 PM" src="https://user-images.githubusercontent.com/67046386/119138268-55424d00-ba5f-11eb-89fb-36fe45a6b689.png">

<img width="326" alt="Screenshot 2021-05-21 at 6 04 31 PM" src="https://user-images.githubusercontent.com/67046386/119138281-583d3d80-ba5f-11eb-885e-08ab341f5ef7.png">.

2. Stable (2.2.0) : mobile:

<img width="339" alt="Screenshot 2021-05-21 at 5 59 29 PM" src="https://user-images.githubusercontent.com/67046386/119138365-6db26780-ba5f-11eb-8a5a-73b2bb26b669.png">

<details>
<summary> flutter doctor -v </summary>

```
[✓] Flutter (Channel stable, 2.2.0, on Mac OS X 10.15.4 19E2269 darwin-x64,
    locale en-GB)
    • Flutter version 2.2.0 at /Users/dhs/documents/fluttersdk/flutter
    • Framework revision b22742018b (5 days ago), 2021-05-14 19:12:57 -0700
    • Engine revision a9d88a4d18
    • Dart version 2.13.0

[!] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    ! CocoaPods 1.9.3 out of date (1.10.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin
        code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see
      https://guides.cocoapods.org/using/getting-started.html#installation for
      instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.55.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (3 available)
    • Darshan's iphone (mobile) • 21150b119064aecc249dfcfe05e259197461ce23 • ios
      • iOS 14.4.1
    • macOS (desktop)           • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)              • chrome                                   •
      web-javascript • Google Chrome 90.0.4430.212

! Doctor found issues in 1 category.


```

</details>

**Comment by darshankawar** [comment]
I am not sure if this is a regression or intended behavior, so not labeling `regression`. 

Cc: @HansMuller

**Comment by kiaxseventh** [comment]
same issue on version 2.2.1

**Comment by kiaxseventh** [comment]
same issue on version 2.2.2

**Comment by HansMuller** [comment]
I'm closing this because, the primaryColor dependency is being phased out for all components, see  https://github.com/flutter/flutter/issues/91772. We've moved on to Material 3 (https://github.com/flutter/flutter/pull/107943) and in doing so InputDecorator and the rest of the Material widgets have been changed to only depend on the colors in the Theme's ColorScheme. The InputDecorator widget renders the text form fields icons and labels as was upgraded to Material 3 in https://github.com/flutter/flutter/pull/107943.

CC @hangyujin @guidezpl
